### PR TITLE
Accept Sets mapping and Xcode 7 beta 4 fix

### PR DESCRIPTION
**Pull request by Busta117**
hi, i see the need of accept Sets mapping for map core data entities, so... look and merge if it looks good

### CODE REVIEW

File: ObjectMapper/Core/Mapper.swift [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 import Foundation
 
-public protocol Mappable {
+public protocol Mappable:Hashable {
```
Comment (ikesyo) [inline ObjectMapper/Core/Mapper.swift]: I believe that `Mappable` must be `Hashable` is a bad decision. Instead, You can add methods below in generic extensions of `Mapper`:

```swift
extension Mapper where N: Hasable {
    public func mapSet(JSONString: String) -> Set<N> { ... }
    ...
}
```

File: ObjectMapper/Core/Operators.swift [hunk 2]
```diff
@@ -345,4 +345,35 @@ public func <- <T: Mappable>(inout left: Array<T>!, right: Map) {
...
 	}
-}
\ No newline at end of file
+}
+
+
+// MARK:- Set of Mappable objects - Set<T: Mappable>
+
+/// Array of Mappable objects
+public func <- <T: Mappable>(inout left: Set<T>, right: Map) {
```
Comment (ikesyo) [inline ObjectMapper/Core/Operators.swift]: These functions should constrain the generic parameter as `<T: Mappable where T: Hashable>`.

File: ObjectMapper/Core/ToJSON.swift [hunk 3]
```diff
@@ -121,7 +121,20 @@ internal final class ToJSON {
             objectArray(field, key: key, dictionary: &dictionary)
         }
     }
-    
+	
+	
+	class func objectSet<N: Mappable>(field: Set<N>, key: String, inout dictionary: [String : AnyObject]) {
```
Comment (ikesyo) [inline ObjectMapper/Core/ToJSON.swift]: These methods should constrain the generic parameter as `<N: Mappable where N: Hashable>`.

**Comment by Busta117** [comment]
fixed with the changes suggested by @ikesyo

**Comment by tristanhimmelman** [comment]
@Busta117 thanks for the addition :+1: 

It would be great if you could add a few tests as well to ensure everything keeps working well.

**Comment by IgorBelyayev** [comment]
I'm unable to successfully deserialize a Set. Has anyone verified this behavior?

**Comment by tristanhimmelman** [comment]
@IgorBelyayev it seems that sets of `Mappable` object function correctly, however, sets of primitive type are not currently working. I will try to add this functionality in the near future

**Pull request merged**